chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -288,68 +284,65 @@ where
             true => (to - from).min(*MAX_EVENT_ONLY_RANGE - 1),
         };
 
-        stream::unfold((from, step), move |(start, step)| {
-            if start > to {
-                return None;
-            }
+        // Typically this will loop only once and fetch the entire range in one request. But if the
+        // node returns an error that signifies the request is to heavy to process, the range will
+        // be broken down to smaller steps.
+        futures03::stream::try_unfold((from, step), move |(start, step)| {
```

Comment:
I checked everything in here carefully for bugs and everything checks out. Nice.

Reply:
Thanks! It's easy to make an off-by-one error here.